docs/docs/porting-a-jekyll-website-to-gatsby.md

```diff
@@ -0,0 +1,276 @@
+
+<!-- The plugin gatsby-transformer-remark is really usefull and also has many plugins to use within it.   -->
+
+## Development environment
+
+Gatsby generates websites and web applications for production through a compilation and build process, and it also has tools optimized for local development. To set up the Gatsby [CLI](/docs/glossary#cli) and development environment (if you haven't already) check out [Part Zero of the Gatsby tutorial](/tutorial/part-zero/).
+
+## Transition to Gatsby
+
+Gatsby can help you set up an application and removes much of the configuration headache. However, Gatsby offers some additional advantages like performance optimizations with static rendering and a thriving ecosystem of plugins, To get things set up and running we need to structure our jekyll-site files like this.
+
+```
+jekyll-site
+  ├── _includes
+  │    ├── footer.html
+  │    └── header.html
+  ├── _layouts
+  │    ├── default.html
+  │    └── post.html
+  ├── _posts
+  │    ├── 2007-10-29-why-every-programmer-should-play-nethack.md
+  │    └── 2009-04-26-barcamp-boston-4-roundup.md
+  ├── _sass
+  ├── _drafts
+  │    ├── begin-with-the-crazy-ideas.md
+  │    └── on-simplicity-in-technology.md
+  ├── _site
+  │    ├── about.md
+  │    └── blah-blah.md
+  ├── script
+  ├── .jekyll-metadata
+  ├── _config.yml
+  └── index.md
+```
+
+### Migrating the markdown / Liquid templates pages
+
+Create a new folder in Gatsby project under the `src/pages-markdown` for those pages written in markdown format. I left the `src/pages` as default so in future we need it for some javaScript files. Now its time to transfer all markdown files to `src/pages-markdown` like this.
+
+```
+  src
+  └── pages-markdown
+      ├── footer.md
+      └── header.md
+```
+
+In addition, these pages must be have next properties within the `frontmatter` section:
+
+```md
+---
+layout: page
+title: About
+date: 2010-11-23 14:48
+path: /about
+---
+```
+
+Now its time to make Gatsby engine take account into these files to inform where to read them:
+
+```json
+// gatsby-config.js
+{
+  "resolve": `gatsby-source-filesystem`,
+  "options": {
+    "name": `markdown-pages`,
+    "path": `${__dirname}/src/pages-markdown/`
+  }
+}
+```
+
+For each file found at `src/pages-markdown` Gatsby will create a `node`. We can extend the `createPages` method and make Gatsby create a new page for each `node`. The steps can be summarized as:
+In addition, these pages must be have next properties within the `frontmatter` section:
+
+- Query for all the markdown nodes read by gatsby
+- Filter those nodes with `layout=page`
+- Create a new page invoking `action.createPage` method and using the template file src/templated/page.js
```

Comment:
This guide hasn't talked about creating a page template. Perhaps that should go first?